feature-libs/asm/customer-360/components/customer-360/customer-360.component.ts

```diff
@@ -0,0 +1,315 @@
+
+  protected showErrorAlert$ = new BehaviorSubject<boolean>(false);
+
+  protected showErrorTab$ = new BehaviorSubject<boolean>(false);
+
+  constructor(
+    protected customer360Config: Customer360Config,
+    protected customer360Facade: Customer360Facade,
+    protected launchDialogService: LaunchDialogService,
+    protected csAgentAuthService: CsAgentAuthService,
+    protected directionService: DirectionService
+  ) {
+    this.customerOverview$ = this.customer360Facade
+      .get360Data([
+        {
+          requestData: { type: Customer360Type.OVERVIEW },
+        },
+      ])
+      .pipe(
+        map((response) => {
+          const overviewItem = response?.value?.find(
+            (item) => item.type === Customer360Type.OVERVIEW
+          ) as Customer360Overview | undefined;
+          return overviewItem?.overview || undefined;
+        }),
+        catchError(() => {
+          this.showErrorAlert$.next(true);
+          return of(undefined);
+        })
+      );
+
+    this.tabs = customer360Config?.customer360?.tabs ?? [];
+    this.currentTab = this.tabs[0];
+  }
+
+  ngOnInit(): void {
+    this.subscription.add(
+      this.csAgentAuthService
+        .isCustomerSupportAgentLoggedIn()
+        .pipe(distinctUntilChanged())
+        .subscribe((loggedIn) => {
+          if (!loggedIn) {
+            this.launchDialogService.closeDialog('logout');
+          }
+        })
+    );
+
+    this.subscription.add(
+      this.launchDialogService.data$.subscribe((data) => {
+        this.customer = data.customer;
+      })
+    );
+    this.setTabData();
+  }
+
+  ngAfterViewInit(): void {
+    this.updateTabIndex(this.activeTab);
+  }
+
+  ngOnDestroy(): void {
+    this.subscription.unsubscribe();
+  }
+
+  get errorAlert$(): Observable<boolean> {
+    return this.showErrorAlert$.asObservable();
+  }
+
+  get errorTab$(): Observable<boolean> {
+    return this.showErrorTab$.asObservable();
+  }
+
+  /**
+   * Triggered when a tab is selected.
+   * @param {number} selectedTab selected tab index
+   */
+  selectTab(selectedTab: number): void {
+    this.activeTab = selectedTab;
+    this.currentTab = this.tabs[selectedTab];
+    this.updateTabIndex(selectedTab);
+    this.setTabData();
+  }
+  /**
+   *  Update tab focus when key is pressed
+   * @param {KeyboardEvent} event
+   * @param {number} selectedIndex current tab index
+   */
+  switchTab(event: KeyboardEvent, selectedIndex: number): void {
+    const maxTab = this.tabs.length - 1;
+    let flag = true;
+    if (this.isBackNavigation(event)) {
+      selectedIndex--;
+      if (selectedIndex < 0) {
+        selectedIndex = maxTab;
+      }
+    } else if (this.isForwardsNavigation(event)) {
+      selectedIndex++;
+      if (selectedIndex > maxTab) {
+        selectedIndex = 0;
+      }
+    } else if (event.code === 'Home') {
+      selectedIndex = 0;
+    } else if (event.code === 'End') {
+      selectedIndex = maxTab;
```

Comment:
[NITPICK] can be fixed anytime
Let's not use hardcoded values in component TS, but rather some shared constants or Enum.